docs/source/installation/development.rst

```diff
@@ -3,5 +3,268 @@
 Installation guide (development)
 ===============================
 
-.. todo::
-    Write guide
+Installing System Packages
+--------------------------
+
+We'll assume you've installed PostgreSQL (and its ``contrib`` package, if on a Linux system) and managed to setup/start
+the server. PostgreSQL 9.6 at least is required.
+
+You can find instructions on how to install it on Fedora/CentOS
+`here <https://wiki.postgresql.org/wiki/YUM_Installation>`_. You can find the same for Debian/Ubuntu
+`here <https://wiki.postgresql.org/wiki/Apt>`_. If you're a macOS user, you will probably want to use Homebrew::
+
+    brew install postgresql
+    brew services start postgresql
+
+
+CentOS/Fedora
++++++++++++++
+
+.. code-block:: shell
+
+    yum install -y gcc redis python-devel python-virtualenv libjpeg-turbo-devel libxslt-devel libxml2-devel \
+        libffi-devel pcre-devel libyaml-devel redhat-rpm-config
+    systemctl start redis.service
+
+
+Debian/Ubuntu
++++++++++++++
+
+.. code-block:: shell
+
+    apt install -y --install-recommends python-dev python-virtualenv libxslt1-dev libxml2-dev libffi-dev libpcre3-dev \
+        libyaml-dev build-essential redis-server
+
+Then on Debian::
+
+    apt install -y libjpeg62-turbo-dev
+
+And on Ubuntu::
+
+    apt install -y libjpeg-turbo8-dev zlib1g-dev
+
+
+macOS
++++++
+
+We recommend that you use `Homebrew <https://brew.sh/>`_::
+
+    brew install python2 redis libjpeg libffi pcre libyaml
+    pip install virtualenv
+
+
+Creating the directory structure
+--------------------------------
+
+You will need a directory in your file system to store Indico as well as its data files (archives, etc...). Some
+developers keep all their code inside a ``dev`` or ``code`` dir. We will assume ``dev`` here.
+
+.. code-block:: shell
+
+    mkdir -p ~/dev/indico/data
+
+We will need a virtualenv where to run Indico::
+
+    cd ~/dev/indico
+    virtualenv env -p /usr/bin/python2.7
+
+
+Cloning Indico
+--------------
+
+First, let's clone Indico's code base. If you're going to contribute back to the project, it's probably best if you
+clone your own `GitHub fork of the project <https://help.github.com/articles/fork-a-repo/>`_ and set it as the origin::
+
+    git clone --recursive git@github.com:<your-github-username>/indico.git src
+    cd src
+    git remote add upstream https://github.com/indico/indico.git src
+    cd ..
+
+Otherwise, cloning the upstream repository as the origin should be enough::
+
+    git clone --recursive https://github.com/indico/indico.git src
+
+If you're going to be changing the standard Indico plugins and/or the documentation, you can also clone those::
+
+    mkdir plugins
```

Comment:
`cd ~/dev/indico` before doing that

Reply:
But... aren't we already there?

Reply:
ah right, didn't notice the `cd ..` above